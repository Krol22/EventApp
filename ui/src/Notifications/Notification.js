import React from 'react';

export const NotificationTypes = {
    SUCCESS: 'SUCCESS',
    ERROR: 'ERROR'
};

export const Notification = ({type, message}) => {
    const classNames = `notification notification--${type.toLowerCase()}`;

    return (
        <div className={classNames}>
            <span className="notification__message">{message}</span>
        </div>
    );
};