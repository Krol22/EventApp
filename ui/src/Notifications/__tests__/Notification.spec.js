import React from 'react';
import { Notification } from '../Notification';
import renderer from 'react-test-renderer';

describe('Notification', () => {
    it('should properly render success notification', () => {
        const successNotificaiton = renderer.create(
            <Notification type="success" message="Your test success message" />
        );

        let tree = successNotificaiton.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should properly render error notification', () => {
        const errorNotification = renderer.create(
            <Notification type="error" message="Your test error message" />
        );

        let tree = errorNotification.toJSON();
        expect(tree).toMatchSnapshot();
    });

});