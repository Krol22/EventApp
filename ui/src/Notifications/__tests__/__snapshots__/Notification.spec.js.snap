// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should properly render error notification 1`] = `
<div
  className="notification notification--error"
>
  <span
    className="notification__message"
  >
    Your test error message
  </span>
</div>
`;

exports[`Notification should properly render success notification 1`] = `
<div
  className="notification notification--success"
>
  <span
    className="notification__message"
  >
    Your test success message
  </span>
</div>
`;
